packages/semcom-upload-component/lib/components/sem-com-register.machine.ts

```diff
@@ -0,0 +1,363 @@
+    quad(namedNode(event.uploadFormContext.uri), namedNode('http://semcom.digita.ai/voc#description'),  literal(event.uploadFormContext.description)),
+    quad(namedNode(event.uploadFormContext.uri), namedNode('http://semcom.digita.ai/voc#author'), literal(event.uploadFormContext.author)),
+    quad(namedNode(event.uploadFormContext.uri), namedNode('http://semcom.digita.ai/voc#version'), literal(event.uploadFormContext.version)),
+    quad(namedNode(event.uploadFormContext.uri), namedNode('http://semcom.digita.ai/voc#latest'), literal(String(event.uploadFormContext.latest))),
+    quad(namedNode(event.uploadFormContext.uri), namedNode('http://semcom.digita.ai/voc#tag'), literal(event.uploadFormContext.tag)),
+    ...shapeQuads,
+  ];
+
+  const writer = new Writer();
+
+  writer.addQuads(quads);
+
+  // Hash the component's uri to serve as the file name
+  const shaObj = new jsSHA('SHAKE256', 'TEXT');
+  shaObj.update(event.uploadFormContext.uri);
+  const hash = shaObj.getHash('B64', { outputLen: 256 }).replace(/[+=/]/g, '').toLowerCase();
+
+  // fetch the contents of the pod
+  return from(solidFetch(context.url, { headers: { 'Accept': 'text/turtle' } })).pipe(
+    switchMap((response) => zip(of(response), from(response.text()))),
+    // return a set of all filenames within the pod
+    map(([ response, responseText ]) => new Set(
+      response.status === 200
+        ? new Parser({ format: 'Turtle' }).parse(responseText)
+          .filter((quadFromResponse) => quadFromResponse.object.value === 'http://www.w3.org/ns/ldp#Resource')
+          .filter((quadFromResponse) => quadFromResponse.subject.value !== '')
+          .map((quadFromResponse) => quadFromResponse.subject.value)
+        : undefined
+    )),
+    // if the pod is offline, return an empty set
+    catchError(() => of(new Set<string>())),
+    switchMap((components) => zip(of(components), of(writer))),
+    switchMap(([ components, quadWriter ]) => {
+
+      // if the component already contains a component with the same name, dont save the data
+      if (components.has(hash)) { return throwError(new Error('Component already exists.')); } else {
+
+        let data = '';
+
+        quadWriter.end((error, result) => {
+
+          if (error) {
+
+            throw error;
+
+          }
+
+          data = result;
+
+        })
+        ;
+
+        // save the metadata of the component to the pod, with the hashed url as the filename
+        return from(solidFetch(context.url, {
+          method: 'POST',
+          headers: {
+            'slug': hash,
+            'content-type': 'text/turtle',
+          },
+          body: data,
+        }));
+
+      }
+
+    }),
+    switchMap((response) => response.status === 201
+      ? of(new DataSavedEvent())
+      : of(new DataNotSavedEvent('Data was not saved succesfully. Please try again.'))),
```

Comment:
Same comment as above, can we make it `map(...)` ?

Reply:
Same comment as above: it has to be an observable.